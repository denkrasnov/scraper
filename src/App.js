import { hot } from "react-hot-loader";
import React, { Component } from "react";

import styles from "./App.css";

class App extends Component {
  static defaultProps = {};

  state = {
    value: "",
    loading: false,
    response: null
  };

  handleChange = event => {
    this.setState({ value: event.target.value });
  };

  handleSubmit = e => {
    e.preventDefault();

    const { value } = this.state;
    this.setState({ loading: true });

    fetch("http://localhost:9001/search", {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({ value })
    })
      .then(response => {
        return response.json();
      })
      .then(myJson => {
        this.setState({ response: JSON.stringify(myJson), loading: false });
      });
  };

  render() {
    const { value, loading, response } = this.state;
    console.log(response);
    return (
      <main className={styles.main}>
        <header className={styles.header}>
          <h1>Compara tehnica</h1>
        </header>
        <div className={styles.underline} />
        <div className={styles.formContainer}>
          {loading ? (
            <p>Loadin...</p>
          ) : (
            <form method="post" onSubmit={this.handleSubmit}>
              <input
                type="text"
                name="search"
                value={value}
                onChange={this.handleChange}
              />
              <button type="submit" value="onSubmit">
                Button
              </button>
            </form>
          )}
        </div>
      </main>
    );
  }
}

export default hot(module)(App);
