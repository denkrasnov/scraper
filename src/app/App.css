.main {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
}

.underline {
  width: 80%;
  height: 1px;
  margin-bottom: 30px;
  background-color: #c8e1ff;
}

.formContainer {
  width: 80%;
  max-width: 550px;
  min-width: 300px;
  background-color: #fff;
  padding: 20px;
  margin-bottom: 45px;
  box-shadow: 0 1px 5px rgba(8, 31, 44, 0.12);
}
