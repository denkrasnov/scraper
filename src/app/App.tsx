import { hot } from "react-hot-loader/root";
import React from "react";

import Home from "./pages/Home";
import ProductsProvider from "./services/fetchProducts";

const App = () => (
  <ProductsProvider>
    <Home />
  </ProductsProvider>
);

// TODO: Move the commented logic below to components

// import Box from "../components/Box";
// import styles from "./App.css";
/* class App extends Component {
  static defaultProps = {};

  // state = {
  //   value: "",
  //   loading: false,
  //   response: null
  // };

  // handleChange = (e: ChangeEvent<HTMLInputElement>) => {
  //   this.setState({ value: e.target.value });
  // };

  // handleSubmit = (e: FormEvent) => {
  //   e.preventDefault();

  //   const { value } = this.state;
  //   this.setState({ loading: true });

  //   fetch("http://localhost:9001/search", {
  //     method: "POST",
  //     headers: {
  //       "Content-Type": "application/json"
  //     },
  //     body: JSON.stringify({ value })
  //   })
  //     .then(response => {
  //       return response.json();
  //     })
  //     .then(myJson => {
  //       this.setState({ response: JSON.stringify(myJson), loading: false });
  //     });
  // };

  render() {
    // const { value, loading, response } = this.state;
    // if (loading) {
    //   return <p>Loading...</p>;
    // }
    return (
      <Home />
      // <main className={styles.main}>
      //   <div className={styles.underline} />
      //   <Box>Heyy</Box>
      //   <div className={styles.formContainer}>
      //     {loading ? (
      //       <p>Loading...</p>
      //     ) : (
      //       <form method="post" onSubmit={this.handleSubmit}>
      //         <input
      //           type="text"
      //           name="search"
      //           value={value}
      //           onChange={this.handleChange}
      //         />
      //         <button type="submit" value="onSubmit">
      //           Button
      //         </button>
      //       </form>
      //     )}
      //   </div>
      // </main>
    );
  }
} */

export default hot(App);
