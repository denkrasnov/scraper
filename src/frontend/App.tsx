import { hot } from "react-hot-loader/root";
import React from "react";
import { ApolloProvider } from "@apollo/client";

import Home from "./pages/Home";
import { client } from "./services/apollo";

const App = () => (
  <ApolloProvider client={client}>
    <Home />
  </ApolloProvider>
);

export default hot(App);
