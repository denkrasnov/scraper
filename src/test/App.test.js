import React from "react";
import { shallow } from "enzyme";

import App from "../App";

const noop = () => {};

describe("App", () => {
  it("renders", () => {
    const component = shallow(<App createTask={noop} loading={false} />);
    expect(component).toMatchSnapshot();
  });

  it("tests the 'createTask'", () => {
    const createTaskSpy = jest.fn();
    const values = { title: "__TITLE__", description: "__DESCRIPTION__" };
    const component = shallow(<App createTask={createTaskSpy} loading />);

    component.instance().createTask(values);
    expect(createTaskSpy).toHaveBeenCalledWith(values);
    expect(component.state().loading).toBe(true);
  });

  it("tests the 'componentDidUpdate'", () => {
    const createTaskSpy = jest.fn();
    const component = shallow(
      <App createTask={createTaskSpy} loading={false} />
    );

    component.state().loading = true;
    component.setProps({ loading: false });
    expect(component.state().loading).toBe(false);
  });
});
