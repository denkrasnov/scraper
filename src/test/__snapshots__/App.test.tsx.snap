// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<main
  className="main"
>
  <Header
    text="Compara technica"
  />
  <div
    className="underline"
  />
  <Box>
    Heyy
  </Box>
  <div
    className="formContainer"
  >
    <form
      method="post"
      onSubmit={[Function]}
    >
      <input
        name="search"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
        value="onSubmit"
      >
        Button
      </button>
    </form>
  </div>
</main>
`;
